### ISSUE TITLE
Trouble paginating grouped query

### ISSUE THREAD

**Issue by jaimeiniesta** [mojotech/scrivener_ecto#17]
Hello, I've found that pagination fails with a query that includes a `group_by`, it looks like the pagination is calculated without doing this grouping, so the number of pages is greater than expected (and the extra pages are therefore empty).

For example, this query without the group_by will result in 9 pages in my project:

```elixir
issues = from(i in Issue,
  where: i.project_id == ^project.id) |> Repo.paginate(params)
```

But if I group the issues, then although only 1 page of results should be available, I still see 9 pages:

```elixir
issues = from(i in Issue,
  where: i.project_id == ^project.id,
  select: %{ sample_id: min(i.id),
             type: i.type,
             sub_type: i.sub_type,
             message: i.message,
             how_many: count("*") },
  group_by: [i.type, i.sub_type, i.message],
  order_by: [:type, desc: count("*")]) |> Repo.paginate(params)
```

**Comment by jaimeiniesta** [comment]
This is using  phoenix 1.2.0, scrivener_ecto 1.1, scrivener_html 1.4

**Comment by drewolson** [comment]
I'm not totally sure what the solution should be here, because right now we have to strip out `group_by` clauses to do counting for total entries and pagination. If you've got ideas, please let me know.

**Comment by jaimeiniesta** [comment]
Thank you, will try to find the way to fix this. In the meantime I'm using scrivener_list to paginate the results instead.

**Comment by drewolson** [comment]
@jaimeiniesta thanks! Take a look at https://github.com/drewolson/scrivener_ecto/issues/14 as well, it may be quasi-related.

**Comment by drewolson** [comment]
I believe 62681ff73954846370ee41047d0a349c530b3d8b will fix this.

**Comment by jaimeiniesta** [comment]
Thank you but I'm afraid this introduced new issues. This query worked OK on scrivener_ecto 1.1.0 but fails in 1.1.2:

```elixir
    sitemaps = from(s in Sitemap,
      join: w in WebPage,
      on: w.sitemap_id == s.id,
      where: [user_id: ^(conn.assigns.current_user.id)],
      group_by: [s.id],
      order_by: [desc: :inserted_at],
      select: { s.id,
                s.url,
                count(w.id),
                sum(w.errors_count),
                sum(w.warnings_count),
                s.inserted_at }) |> Repo.paginate(params)
```

The error message says:

`** (Ecto.QueryError) subquery can only select sources (t) or fields (t.field), got: count(&1.id()) in query:`

And if I removed this `count(w.id)` from the `select`, then:

`** (Ecto.QueryError) subquery can only select sources (t) or fields (t.field), got: sum(&1.errors_count()) in query:`

**Comment by drewolson** [comment]
I'm pushing a fix for this now, will be released soon. Thanks for the report!

**Comment by drewolson** [comment]
This is now fixed in `1.1.3` and released.
